docs/azure/dependency-injection.md

```diff
@@ -0,0 +1,233 @@
+{
+  "AzureDefaults": {
+    "Diagnostics": {
+      "IsTelemetryDisabled": false,
+      "IsLoggingContentEnabled": true
+    },
+    "Retry": {
+      "MaxRetries": 3,
+      "Mode": "Exponential"
+    }
+  },
+  "KeyVault": {
+    "VaultUri": "https://mykeyvault.vault.azure.net"
+  },
+  "Storage": {
+    "ServiceUri": "https://mydemoaccount.storage.windows.net"
+  }
+}
+```
+
+You can add any options from the [ClientOptions](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/ClientOptions.cs) into the `AzureDefaults` section. One of the options is the retry policy. See the section [Configure a new try policy](#configure-a-new-retry-policy).
+
+Since the `Configuration` object is injected from the host and stored inside the `Startup` constructor, you can do the following:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+  services.AddAzureClients(builder =>
+  {
+    // Add a KeyVault client
+    builder.AddSecretClient(Configuration.GetSection("KeyVault"));
+
+    // Add a storage account client
+    builder.AddBlobServiceClient(Configuration.GetSection("Storage"));
+
+    // Use DefaultAzureCredential by default
+    builder.UseCredential(new DefaultAzureCredential());
+
+    // Set up any default settings
+    builder.ConfigureDefaults(Configuration.GetSection("AzureDefaults"));
+  });
+
+  services.AddControllers();
+}
+```
+
+## Configure multiple service clients with different names
+
+Say you have two storage accounts – one for private information and one for public information. Your application transfers data from the public to private storage account after some operation. You need to have two storage service clients. To set this up in `Startup.ConfigureServices`:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+  services.AddAzureClients(builder =>
+  {
+    builder.AddBlobServiceClient(Configuration.GetSection("PublicStorage"));
+    builder.AddBlobServiceClient(Configuration.GetSection("PrivateStorage"))
+      .WithName("PrivateStorage");
+  });
+}
+```
+
+In your controllers, you can access the named service clients using the `IAzureClientFactory`:
+
+```csharp
+public class HomeControllers : Controller
+{
+  private BlobServiceClient publicStorage, privateStorage;
+
+  public HomeController(BlobServiceClient defaultClient, IAzureClientFactory<BlobServiceClient> clientFactory)
+  {
+    this.publicStorage = defaultClient;
+    this.privateStorage = clientFactory.GetClient("PrivateStorage");
+  }
+}
+```
+
+The un-named service client is still available in the same way as before. Named clients are additive to this.
+
+## Configure a new retry policy
+
+At some point, you might want to change the default settings for a service client. You may want different retry settings or to use a different service API version, for example. You can set the retry settings globally or on a per service basis. Say you have the following `appsettings.json` file:
```

Comment:
```suggestion
At some point, you might want to change the default settings for a service client. You may want different retry settings or to use a different service API version, for example. You can set the retry settings globally or on a per service basis. Say you have the following _appsettings.json_ file:
```

Referencing a file name should be italic, instead of code fenced - per MS style guidelines.